### PR TITLE
Replace `Debug.Assert(false,` with `Debug.Fail(`

### PR DESCRIPTION
**Pull request by stephentoub**

### CODE REVIEW

File: src/libraries/System.Diagnostics.Tracing/tests/BasicEventSourceTest/EventSourceTestParser.cs [hunk 1]
```diff
@@ -658,7 +658,7 @@
 //                case 0:
 //                    return x;
 //                default:
-//                    Debug.Assert(false, "Bad field index");
+//                    Debug.Fail("Bad field index");
```
Comment (danmoseley) [inline src/libraries/System.Diagnostics.Tracing/tests/BasicEventSourceTest/EventSourceTestParser.cs:661]: this comment says "This code was automatically generated by the TraceParserGen tool" .. unless that tool is updated, whereever it is, the comment probably shouldn't change.

**Review by danmoseley — APPROVED** [review]

File: src/libraries/System.Diagnostics.Tracing/tests/BasicEventSourceTest/EventSourceTestParser.cs (hunk 1 above)
Comment (stephentoub) [reply src/libraries/System.Diagnostics.Tracing/tests/BasicEventSourceTest/EventSourceTestParser.cs:661]: Seems like the whole file could just be deleted. @brianrob?

Comment (brianrob) [reply src/libraries/System.Diagnostics.Tracing/tests/BasicEventSourceTest/EventSourceTestParser.cs:661]: Looks like it's all comments, so yes.  @davmason?

Comment (stephentoub) [reply src/libraries/System.Diagnostics.Tracing/tests/BasicEventSourceTest/EventSourceTestParser.cs:661]: https://github.com/dotnet/runtime/pull/105237

Comment (brianrob) [reply src/libraries/System.Diagnostics.Tracing/tests/BasicEventSourceTest/EventSourceTestParser.cs:661]: 👍Just found it as I catch up on e-mail.

**Pull request merged**